gtsam/hybrid/DCFactorGraph.h

```diff
@@ -34,10 +34,7 @@ class DCFactorGraph : public gtsam::FactorGraph<DCFactor> {
       if (factor) {
         // Insert all the discrete keys to the final result.
         for (auto&& key : factor->discreteKeys()) {
-          // Check if key doesn't already exist. If it doesn't then add it.
-          if (std::find(result.begin(), result.end(), key) == result.end()) {
```

Comment:
What happens if you have duplicate keys?

Reply:
That is taken care of by the factors themselves.